rcl_yaml_param_parser/include/rcl_yaml_param_parser/types.h

```diff
@@ -74,12 +74,39 @@ typedef struct rcl_node_params_s
   size_t num_params;  ///< Number of parameters in the node
 } rcl_node_params_t;
 
+/// \typedef rcl_param_descriptor_t
+/// \brief param_descriptor_t stores the descriptor of a parameter
+typedef struct rcl_param_descriptor_s
+{
+  char * name;
+  bool * read_only;
+  uint8_t * type;
+  char * description;
+  char * additional_constraints;
+  double * min_value_double;
+  double * max_value_double;
+  double * step_double;
+  int64_t * min_value_int;
+  int64_t * max_value_int;
+  int64_t * step_int;
+} rcl_param_descriptor_t;
```

Comment:
@wjwwood meta: not a comment for you @bpwilcox, but replicating `rcl_interfaces/msg/ParameterDescriptor` structure is not ideal. The same goes with `rcl_interfaces/msg/Parameter`, so I wonder if we should be re-using the message C representation instead (like we partially do in [`rcl_action`](https://github.com/ros2/rcl/blob/9034c406c2d5bc4ec291f8e1ad4303cde0184cbc/rcl_action/include/rcl_action/types.h#L65-L80) already). Thoughts?

Reply:
In a related note, I understand pointers are used here to have indication of absence. At the same time, ROS 2 interfaces have no optional fields, so something, somewhere, will choose defaults. If we had all of that here, we could get rid of many heap allocations and simplify the code.